util/test/Test/Pos/Util/Base16.hs

```diff
@@ -58,3 +66,32 @@ chunkBS n xs = case LB.uncons xs of
     Just _  ->
         let (taken, dropped) = LB.splitAt n xs
         in  taken : chunkBS n dropped
+
+----------------------------------------------------------------------------
+-- Decoding
+----------------------------------------------------------------------------
+
+-- | Decode a given ByteString which was originally encoded using 'encode' or
+-- 'encodeWithIndex'.
+decode :: LB.ByteString -> LB.ByteString
+decode bs
+    -- No complex parsing is required for data whose length is <= 32.
+    | LB.length bs <= lineWrapLength = fst $ B16.decode bs
+    | otherwise = case PLB.eitherResult $ PLB.parse decodeParser bs of
+        Left l  -> error $ fromString l
```

Comment:
@erikd I know this "error" sucks, but I'm unsure as to what course of action I should take in this situation. I could return an empty ByteString, but then that doesn't provide any information about what happened. 

I suppose one option would be to return an "Either SomeDecodeErrorType LB.ByteString". What do you think?